src/main/webapp/app/entities/course/course.component.html

```diff
@@ -133,14 +133,7 @@ <h4 id="course-page-heading" jhiTranslate="artemisApp.course.home.title">Courses
                                         <fa-icon [icon]="'pencil-alt'"></fa-icon>
                                         <span class="d-none d-md-inline">{{ 'entity.action.edit' | translate }}</span>
                                     </button>
-                                    <button
-                                        type="submit"
-                                        *jhiHasAnyAuthority="['ROLE_ADMIN']"
-                                        [routerLink]="['/', { outlets: { popup: 'course/' + course.id + '/delete' } }]"
-                                        replaceUrl="true"
-                                        queryParamsHandling="merge"
-                                        class="btn btn-danger btn-sm mr-1 mb-1"
-                                    >
+                                    <button type="submit" *jhiHasAnyAuthority="['ROLE_ADMIN']" (click)="openDeleteCourseDialog(course)" class="btn btn-danger btn-sm mr-1 mb-1">
```

Comment:
Please consider introducing a component for this button. This would allow to remove duplicate logic for opening the modal in the component. Only this new component should be needed to know about the `DeleteDialogService`.

Reply:
Create `DeleteButtonComponent`. It does remove some duplicate code already, but I still have some styling issues, because button is now wrapped in Angular component and can't be grouped by Bootstrap `button-group`

Reply:
Maybe this could be solved with a HostBinding: https://angular.io/api/core/HostBinding

Reply:
So I found out that this is an actual problem that is not solved properly for Angular https://github.com/angular/angular/issues/18877
Although, there is a [hack](https://stackoverflow.com/questions/37846473/make-a-button-group-out-of-directives-components), but I would not use it. So for now I am not sure how to proceed better with this one. Would be the solution to layout buttons differently without using `button-group` @jpbernius @thilo-behnke ?

Reply:
I suggest to ignore this issue for now to get the PR merged. It's not ideal but this issue will likely come up again and we can do a small PR to fix it later.